### PR TITLE
Add Pocket Browser to flathub

### PR DESCRIPTION
**Pull request by AhmadHrb**

### CODE REVIEW

File: ml.pocketinc.pocketbrowser.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+base-version: stable
+command: install.sh
+finish-args:
+  - --share=ipc
+  - --socket=x11
+  - --socket=pulseaudio
+  - --share=network
+  build-options:
+  cflags: -O2 -g
+  cxxflags: -O2 -g
```
Comment (barthalion) [inline ml.pocketinc.pocketbrowser.yml:16]:
```suggestion
```

File: ml.pocketinc.pocketbrowser.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
+app-id: ml.pocketinc.pocketbrowser
+runtime: org.freedesktop.Platform
+runtime-version: '1.6'
```
Comment (barthalion) [inline ml.pocketinc.pocketbrowser.yml:3]:
```suggestion
runtime-version: '20.08'
```

File: ml.pocketinc.pocketbrowser.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+branch: stable
+sdk: org.freedesktop.Sdk
+base: io.atom.electron.BaseApp
+base-version: stable
```
Comment (barthalion) [inline ml.pocketinc.pocketbrowser.yml:7]:
```suggestion
base-version: 20.08
```

File: ml.pocketinc.pocketbrowser.yml [hunk 4]
```diff
@@ -0,0 +1,44 @@
+app-id: ml.pocketinc.pocketbrowser
+runtime: org.freedesktop.Platform
+runtime-version: '1.6'
+branch: stable
+sdk: org.freedesktop.Sdk
+base: io.atom.electron.BaseApp
```
Comment (barthalion) [inline ml.pocketinc.pocketbrowser.yml:6]:
```suggestion
base: org.electronjs.Electron2.BaseApp 
```

File: install.sh [hunk 5]
```diff
@@ -0,0 +1,8 @@
+build-commands:
```
Comment (barthalion) [inline install.sh:1]: I assume you wanted to add it to the main manifest?

Comment (AhmadHrb) [reply install.sh:1]: This is my first flathub application. I didn't know what I should add to the install.sh

**Comment by nedrichards** [comment]
bot, build  ml.pocketinc.pocketbrowser

**Comment by nedrichards** [comment]
bot, build ml.pocketinc.pocketbrowser

File: ml.pocketinc.pocketbrowser.yml [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+  commands:
+    - npm start --prefix=/app/main
+build-commands:
+  # Install npm dependencies
+  - npm install --prefix=main --offline --cache=/run/build/pocket-browser/npm-cache/
+  # Bundle app and dependencies
+  - mkdir -p /app/main /app/bin
+  - cp -ra main/* /app/main/
+  # Install app wrapper
+  - install install.sh /app/bin/
```
Comment (barthalion) [inline ml.pocketinc.pocketbrowser.yml:49]: This is misindented by one level.

**Comment by AhmadHrb** [comment]
bot, build ml.pocketinc.pocketbrowser

**Comment by AhmadHrb** [comment]
bot, build ml.pocketinc.pocketbrowser

**Comment by AhmadHrb** [comment]
bot, build ml.pocketinc.pocketbrowser

**Comment by AhmadHrb** [comment]
bot, build ml.pocketinc.pocketbrowser

**Comment by AhmadHrb** [comment]
bot, build ml.pocketinc.pocketbrowser

**Comment by AhmadHrb** [comment]
what should run.sh include?
I didn't find any information about it..

**Comment by AhmadHrb** [comment]
bot, build ml.pocketinc.pocketbrowser

**Comment by AhmadHrb** [comment]
bot, build ml.pocketinc.pocketbrowser

**Comment by AhmadHrb** [comment]
bot, build ml.pocketinc.pocketbrowser

**Comment by AhmadHrb** [comment]
bot, build ml.pocketinc.pocketbrowser

**Comment by AhmadHrb** [comment]
bot, build ml.pocketinc.pocketbrowser

**Comment by AhmadHrb** [comment]
okay I don't know how to fix this.

**Comment by barthalion** [comment]
```diff
diff --git a/ml.pocketinc.pocketbrowser.yml b/ml.pocketinc.pocketbrowser.yml
index ca2605d..609ae1b 100644
--- a/ml.pocketinc.pocketbrowser.yml
+++ b/ml.pocketinc.pocketbrowser.yml
@@ -4,41 +4,39 @@ runtime-version: '20.08'
 branch: stable
 sdk: org.freedesktop.Sdk
 base: org.electronjs.Electron2.BaseApp
-base-version: 20.08
+base-version: '20.08'
 command: run.sh
 finish-args:
   - --share=ipc
   - --socket=x11
   - --socket=pulseaudio
   - --share=network
+
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.node12
+
 modules:
-  - name: nodejs
-    sources:
-      - type: archive
-        url: https://nodejs.org/dist/v12.0.0/node-v12.0.0.tar.xz
-        sha256: 93c48550e5da71cf3809ba765e8f8c01124a6b5f8b69869fe3ea8f168fcbfc78
   - name: pocket-browser
-    build-options:
-      env:
-        # Need this for electron-download to find the cached electron binary
-        electron_config_cache: /run/build/pocket-browser/npm-cache
     buildsystem: simple
+    build-options:
+      append-path: /usr/lib/sdk/node12/bin
+      env:
+        electron_config_cache: /run/build/pocket-browser/npm-cache/cache
+        npm_config_nodedir: /usr/lib/sdk/node12
+    build-commands:
+      - npm install --prefix=pocketbrowser --offline --cache=/run/build/pocket-browser/npm-cache/
+      - mkdir -p /app/main /app/bin
+      - cp -ra main/* /app/main/
+      - install run.sh /app/bin/
     sources:
       - type: archive
         url: https://github.com/PocketInc/Pocket-Browser/archive/v1.5.tar.gz
         sha256: a224ad547663d416c6bbd79364636a6c67101fb7595be2c2b0e8445d89a5f5f0
-        dest: main
+        dest: pocketbrowser
       - generated-sources.json
-      # Wrapper to launch the app
+
       - type: script
         dest-filename: run.sh
         commands:
           - npm start --prefix=/app/main
-    build-commands:
-      # Install npm dependencies
-      - npm install --prefix=main --offline --cache=/run/build/pocket-browser/npm-cache/
-      # Bundle app and dependencies
-      - mkdir -p /app/main /app/bin
-      - cp -ra main/* /app/main/
-      # Install app wrapper
-      - install run.sh /app/bin/
+
```

This will be a good start. It still fails but I guess due to some mistake in package-lock.json that makes some packages missing from generated-sources file.

**Comment by AhmadHrb** [comment]
Thank you for your help! I pushed the improvements.. hoping that would cause it to work?

**Comment by AhmadHrb** [comment]
bot, build ml.pocketinc.pocketbrowser

**Comment by AhmadHrb** [comment]
bot, build ml.pocketinc.pocketbrowser

**Comment by AhmadHrb** [comment]
Okay... It passed the building step so i see improvements..

**Comment by barthalion** [comment]
Yup, you need a desktop and metainfo files. The latter can be generated with https://www.freedesktop.org/software/appstream/metainfocreator/#/ – it won't pass validation but it's a good starting point.

**Pull request closed without merging**